### ISSUE TITLE
Regression on powerpc64 little and big endian

### ISSUE THREAD

**Issue by cohomology** [zherczeg/sljit#130]
A unit test of mine went red after I switched from the sljit version of pcre 10.36 to current trunc. The test only fails on power pc (linuxppc64 little endian and AIX big endian). The unit test still works on amd64.

The jitted function is a function which returns an integer 1 ( = true ) or 0 (false). On ppc64 with the new sljit version, the unit test returns true instead of false.

In our interpreter we are executing a COLUMN <> VARIABLE inside a database engine. At the end memcmp is called to do the actual comparison. This is what the function should do. As both COLUMN and VARIABLE are identical, the expected result is FALSE.

The code is generated very easily. We have a stack machine with instructions:

PUSH_COLUMN
PUSH_PARAMETER
NOT_EQUALS

- The table lines are transferred via SLJIT_S0. In this example we only have a single table.
- SLJIT_S1 is a stack which we allocate outside of the jitted coding in C/C++. 
- The parameters are directly "baked" into the Jit coding. In the below case we work with characters, so we write pointers on the stack (for columns and parameters).

Below is the C++ coding for the working and not-working case. As there were API breaks the C++ coding can not be the same between both versions.

The old version ok.cpp (worked everytime on any platform, old version of sljit):

```cpp

#include "sljitLir.h"
#include <iostream>
#include <cstring>

static void dump_code(void* code, sljit_uw len)
{
  FILE *fp = fopen("/tmp/slj_dump", "wb");
  if(!fp)
    return;
  fwrite(code, len, 1, fp);
  fclose(fp);
#if defined(__x86_64__)
  system("objdump -b binary -m l1om -D /tmp/slj_dump");
#else
  system("objdump -b binary -m powerpc --endian=little -M power9 -D /tmp/slj_dump");
#endif
}

struct Indirection
{
  const char * string;
};

int mymemcmp(void * a, void * b, unsigned long length)
{
  int result = memcmp(a, b, length);
  std::cout << "Result of memcmp: " << result << std::endl;
  return result;
}

int main()
{
  const char * string1 = "ABCHello     ";
  const char * string2 = "Hello     ";

  Indirection indirection { string1 };

  sljit_compiler * compiler =
    sljit_create_compiler(nullptr, nullptr);

  sljit_emit_enter(compiler, 0, SLJIT_ARG1(SW) | SLJIT_ARG2(SW),
                   5 /* scratch register: R0, R1, R2, R3 */,
                   3 /* saved registers, S0, S1, S2 */,
                   2 /* scratch floating point registers, FR0 */,
                   0 /* saved floating point registers */,
                   0 /* no local stack */);
  sljit_emit_op1(compiler, SLJIT_MOV, SLJIT_R0, 0,
    SLJIT_MEM1(SLJIT_S0), offsetof(Indirection, string));
  sljit_emit_op2(compiler, SLJIT_ADD, SLJIT_R0, 0,
      SLJIT_R0, 0, SLJIT_IMM, 3);
  sljit_emit_op1(compiler, SLJIT_MOV, SLJIT_MEM1(SLJIT_S1), 0,
      SLJIT_R0, 0);
  sljit_emit_op2(compiler, SLJIT_ADD, SLJIT_S1, 0,
      SLJIT_S1, 0, SLJIT_IMM, sizeof(sljit_sw));
  sljit_emit_op1(compiler, SLJIT_MOV, SLJIT_MEM1(SLJIT_S1), 0,
      SLJIT_IMM, (sljit_sw) string2);
  sljit_emit_op2(compiler, SLJIT_ADD, SLJIT_S1, 0,
      SLJIT_S1, 0, SLJIT_IMM, sizeof(sljit_sw));

  sljit_emit_op1(compiler, SLJIT_MOV, SLJIT_R0, 0,
      SLJIT_MEM1(SLJIT_S1), - 2 * sizeof(sljit_sw));
  sljit_emit_op1(compiler, SLJIT_MOV, SLJIT_R1, 0,
      SLJIT_MEM1(SLJIT_S1), -  sizeof(sljit_sw));
  sljit_emit_op1(compiler, SLJIT_MOV, SLJIT_R2, 0,
      SLJIT_IMM, static_cast<sljit_sw>(10));
  sljit_emit_icall(compiler, SLJIT_CALL,
      SLJIT_RET(SW) | SLJIT_ARG1(SW) | SLJIT_ARG2(SW) | SLJIT_ARG3(SW),
      SLJIT_IMM, SLJIT_FUNC_OFFSET(mymemcmp));

  sljit_emit_op1(compiler, SLJIT_MOV, SLJIT_R1, 0,
      SLJIT_IMM, 0);

  sljit_emit_op2(compiler, SLJIT_OR | SLJIT_SET_Z, SLJIT_UNUSED, 0,
      SLJIT_RETURN_REG, 0, SLJIT_IMM, 0);

  sljit_emit_cmov(compiler, SLJIT_ZERO, SLJIT_R1, SLJIT_IMM, 1);

  sljit_emit_op1(compiler, SLJIT_MOV, SLJIT_MEM1(SLJIT_S1), - 2 * sizeof(sljit_sw),
      SLJIT_R1, 0);

  sljit_emit_op2(compiler, SLJIT_SUB, SLJIT_S1, 0,
      SLJIT_S1, 0, SLJIT_IMM, sizeof(sljit_sw));

  sljit_jump * jumpIfPassed =
    sljit_emit_cmp(compiler, SLJIT_EQUAL,
                   SLJIT_MEM1(SLJIT_S1), - sizeof(sljit_sw),
                   SLJIT_IMM, 0);

  sljit_emit_return(compiler, SLJIT_MOV, SLJIT_IMM, 0);

  sljit_set_label(jumpIfPassed,
    sljit_emit_label(compiler));

  sljit_emit_return(compiler, SLJIT_MOV, SLJIT_IMM, 1);

  void * code = sljit_generate_code(compiler);

  dump_code(code,
            sljit_get_generated_code_size(compiler));

  sljit_sw * stack = (sljit_sw*) malloc(sizeof(sljit_sw) * 20);

  typedef sljit_sw (*func_t)(void * line, void * stack);

  func_t func = (func_t) code;

  sljit_sw result = func(&indirection, stack);

  std::cout << result << std::endl;

  free(stack);

  return 0;
}
```

With the new sljit, we get different results on PPC64 and AMD64. Only the AMD64 version yields the same result as the ok.cpp version above (written with an old version of sljit).

```cpp
#include "sljitLir.h"
#include <cstring>
#include <iostream>

static void dump_code(void* code, sljit_uw len)
{
  FILE *fp = fopen("/tmp/slj_dump", "wb");
  if(!fp)
    return;
  fwrite(code, len, 1, fp);
  fclose(fp);
#if defined(__x86_64__)
  system("objdump -b binary -m l1om -D /tmp/slj_dump");
#else
  system("objdump -b binary -m powerpc --endian=little -M power9 -D /tmp/slj_dump");
#endif
}

struct Indirection
{
  const char * string;
};

int mymemcmp(void * a, void * b, unsigned long length)
{
  int result = memcmp(a, b, length);
  std::cout << "Result of memcmp: " << result << std::endl;
  return result;
}

int main()
{
  const char * string1 = "ABCHello     ";
  const char * string2 = "Hello     ";

  Indirection indirection { string1 };

  sljit_compiler * compiler =
    sljit_create_compiler(nullptr, nullptr);

  sljit_emit_enter(compiler, 0, SLJIT_ARGS2(W, W, W),
                   5 /* scratch register: R0, R1, R2, R3 */,
                   3 /* saved registers, S0, S1, S2 */,
                   2 /* scratch floating point registers, FR0 */,
                   0 /* saved floating point registers */,
                   0 /* no local stack */);

  sljit_emit_op1(compiler, SLJIT_MOV, SLJIT_R0, 0,
    SLJIT_MEM1(SLJIT_S0),  offsetof(Indirection, string));
  sljit_emit_op2(compiler, SLJIT_ADD, SLJIT_R0, 0,
      SLJIT_R0, 0, SLJIT_IMM, 3);
  sljit_emit_op1(compiler, SLJIT_MOV, SLJIT_MEM1(SLJIT_S1), 0,
      SLJIT_R0, 0);
  sljit_emit_op2(compiler, SLJIT_ADD, SLJIT_S1, 0,
      SLJIT_S1, 0, SLJIT_IMM, sizeof(sljit_sw));
  sljit_emit_op1(compiler, SLJIT_MOV, SLJIT_MEM1(SLJIT_S1), 0,
      SLJIT_IMM, (sljit_sw) string2);
  sljit_emit_op2(compiler, SLJIT_ADD, SLJIT_S1, 0,
      SLJIT_S1, 0, SLJIT_IMM, sizeof(sljit_sw));

  sljit_emit_op1(compiler, SLJIT_MOV, SLJIT_R0, 0,
      SLJIT_MEM1(SLJIT_S1), - 2 * sizeof(sljit_sw));
  sljit_emit_op1(compiler, SLJIT_MOV, SLJIT_R1, 0,
      SLJIT_MEM1(SLJIT_S1), -  sizeof(sljit_sw));
  sljit_emit_op1(compiler, SLJIT_MOV, SLJIT_R2, 0,
      SLJIT_IMM, static_cast<sljit_sw>(10));
  sljit_emit_icall(compiler, SLJIT_CALL,
      SLJIT_ARGS3(W, W, W, W),
      SLJIT_IMM, SLJIT_FUNC_ADDR(mymemcmp));

  sljit_emit_op1(compiler, SLJIT_MOV, SLJIT_R1, 0,
      SLJIT_IMM, 0);

  sljit_emit_op2u(compiler, SLJIT_OR | SLJIT_SET_Z,
      SLJIT_RETURN_REG, 0, SLJIT_IMM, 0);

  sljit_emit_cmov(compiler, SLJIT_ZERO, SLJIT_R1, SLJIT_IMM, 1);

  sljit_emit_op1(compiler, SLJIT_MOV, SLJIT_MEM1(SLJIT_S1), - 2 * sizeof(sljit_sw),
      SLJIT_R1, 0);

  sljit_emit_op2(compiler, SLJIT_SUB, SLJIT_S1, 0,
      SLJIT_S1, 0, SLJIT_IMM, sizeof(sljit_sw));

  sljit_jump * jumpIfPassed =
    sljit_emit_cmp(compiler, SLJIT_EQUAL,
                   SLJIT_MEM1(SLJIT_S1), - sizeof(sljit_sw),
                   SLJIT_IMM, 0);

  sljit_emit_return(compiler, SLJIT_MOV, SLJIT_IMM, 0);

  sljit_set_label(jumpIfPassed,
    sljit_emit_label(compiler));

  sljit_emit_return(compiler, SLJIT_MOV, SLJIT_IMM, 1);

  void * code = sljit_generate_code(compiler);

  dump_code(code,
            sljit_get_generated_code_size(compiler));

  sljit_sw * stack = (sljit_sw*) malloc(sizeof(sljit_sw) * 20);

  typedef sljit_sw (*func_t)(void * line, void * stack);

  func_t func = (func_t) code;

  sljit_sw result = func(&indirection, stack);

  std::cout << result << std::endl;

  free(stack);

  return 0;
}
```

The generated assembly on power pc:

ok.cpp (old version of sljit)
```
   0:   a6 02 08 7c     mflr    r0
   4:   f8 ff e1 fb     std     r31,-8(r1)
   8:   f0 ff c1 fb     std     r30,-16(r1)
   c:   e8 ff a1 fb     std     r29,-24(r1)
  10:   e0 ff 81 fb     std     r28,-32(r1)
  14:   10 00 01 f8     std     r0,16(r1)
  18:   00 00 e0 3b     li      r31,0
  1c:   78 1b 7e 7c     mr      r30,r3
  20:   78 23 9d 7c     mr      r29,r4
  24:   71 ff 21 f8     stdu    r1,-144(r1)
  28:   00 00 7e e8     ld      r3,0(r30)
  2c:   03 00 63 38     addi    r3,r3,3
  30:   00 00 7d f8     std     r3,0(r29)
  34:   08 00 bd 3b     addi    r29,r29,8
  38:   01 10 40 3d     lis     r10,4097
  3c:   50 95 4a 61     ori     r10,r10,38224
  40:   00 00 5d f9     std     r10,0(r29)
  44:   08 00 bd 3b     addi    r29,r29,8
  48:   f0 ff 7d e8     ld      r3,-16(r29)
  4c:   f8 ff 9d e8     ld      r4,-8(r29)
  50:   0a 00 a0 38     li      r5,10
  54:   00 10 80 3d     lis     r12,4096
  58:   fc 12 8c 61     ori     r12,r12,4860
  5c:   a6 03 89 7d     mtctr   r12
  60:   21 1c 80 4e     bctrl   3
  64:   78 fb e4 7f     mr      r4,r31
  68:   79 fb 6a 7c     or.     r10,r3,r31
  6c:   08 00 82 40     bne     0x74
  70:   01 00 80 38     li      r4,1
  74:   f0 ff 9d f8     std     r4,-16(r29)
  78:   f8 ff bd 3b     addi    r29,r29,-8
  7c:   f8 ff 3d e9     ld      r9,-8(r29)
  80:   00 f8 29 7c     cmpd    r9,r31
  84:   28 00 82 41     beq     0xac
  88:   78 fb e3 7f     mr      r3,r31
  8c:   90 00 21 38     addi    r1,r1,144
  90:   10 00 01 e8     ld      r0,16(r1)
  94:   e0 ff 81 eb     ld      r28,-32(r1)
  98:   e8 ff a1 eb     ld      r29,-24(r1)
  9c:   f0 ff c1 eb     ld      r30,-16(r1)
  a0:   f8 ff e1 eb     ld      r31,-8(r1)
  a4:   a6 03 08 7c     mtlr    r0
  a8:   20 00 80 4e     blr
  ac:   01 00 60 38     li      r3,1
  b0:   90 00 21 38     addi    r1,r1,144
  b4:   10 00 01 e8     ld      r0,16(r1)
  b8:   e0 ff 81 eb     ld      r28,-32(r1)
  bc:   e8 ff a1 eb     ld      r29,-24(r1)
  c0:   f0 ff c1 eb     ld      r30,-16(r1)
  c4:   f8 ff e1 eb     ld      r31,-8(r1)
  c8:   a6 03 08 7c     mtlr    r0
  cc:   20 00 80 4e     blr
```

fail.cpp (new version of sljit):

```
   0:   a6 02 08 7c     mflr    r0
   4:   71 ff 21 f8     stdu    r1,-144(r1)
   8:   88 00 e1 fb     std     r31,136(r1)
   c:   80 00 c1 fb     std     r30,128(r1)
  10:   78 00 a1 fb     std     r29,120(r1)
  14:   70 00 81 fb     std     r28,112(r1)
  18:   a0 00 01 f8     std     r0,160(r1)
  1c:   00 00 e0 3b     li      r31,0
  20:   78 1b 7e 7c     mr      r30,r3
  24:   78 23 9d 7c     mr      r29,r4
  28:   00 00 7e e8     ld      r3,0(r30)
  2c:   03 00 63 38     addi    r3,r3,3
  30:   00 00 7d f8     std     r3,0(r29)
  34:   08 00 bd 3b     addi    r29,r29,8
  38:   01 10 40 3d     lis     r10,4097
  3c:   28 b2 4a 61     ori     r10,r10,45608
  40:   00 00 5d f9     std     r10,0(r29)
  44:   08 00 bd 3b     addi    r29,r29,8
  48:   f0 ff 7d e8     ld      r3,-16(r29)
  4c:   f8 ff 9d e8     ld      r4,-8(r29)
  50:   0a 00 a0 38     li      r5,10
  54:   00 10 80 3d     lis     r12,4096
  58:   bc 12 8c 61     ori     r12,r12,4796
  5c:   a6 03 89 7d     mtctr   r12
  60:   21 1c 80 4e     bctrl   3
  64:   78 fb e4 7f     mr      r4,r31
  68:   00 00 6a 60     ori     r10,r3,0
  6c:   00 00 4a 65     oris    r10,r10,0
  70:   08 00 82 40     bne     0x78
  74:   01 00 80 38     li      r4,1
  78:   f0 ff 9d f8     std     r4,-16(r29)
  7c:   f8 ff bd 3b     addi    r29,r29,-8
  80:   f8 ff 3d e9     ld      r9,-8(r29)
  84:   00 00 29 2c     cmpdi   r9,0
  88:   28 00 82 41     beq     0xb0
  8c:   78 fb e3 7f     mr      r3,r31
  90:   a0 00 01 e8     ld      r0,160(r1)
  94:   88 00 e1 eb     ld      r31,136(r1)
  98:   80 00 c1 eb     ld      r30,128(r1)
  9c:   78 00 a1 eb     ld      r29,120(r1)
  a0:   70 00 81 eb     ld      r28,112(r1)
  a4:   a6 03 08 7c     mtlr    r0
  a8:   90 00 21 38     addi    r1,r1,144
  ac:   20 00 80 4e     blr
  b0:   01 00 60 38     li      r3,1
  b4:   a0 00 01 e8     ld      r0,160(r1)
  b8:   88 00 e1 eb     ld      r31,136(r1)
  bc:   80 00 c1 eb     ld      r30,128(r1)
  c0:   78 00 a1 eb     ld      r29,120(r1)
  c4:   70 00 81 eb     ld      r28,112(r1)
  c8:   a6 03 08 7c     mtlr    r0
  cc:   90 00 21 38     addi    r1,r1,144
  d0:   20 00 80 4e     blr
```

**Comment by cohomology** [comment]
Sorry we can't really point it down. 

The above assembly output is from the demo program which we wrote to reproduce the example. 

We completed the demo program that it also calls the jitted function and in the *demo program* it seems to work now unfortunately. So it only fails in our productive program.

What we did in our productive program now was to change the called function (above it is called myfunc) to a local function which simply "wraps" the other function. Then it started to work.

Could it be that the exact address of the called function is important? Or is it some uninitialized variable or other strange thing?

In any case, I know extracted the "bad" assembly generated in my productive program:

```
   0:   a6 02 08 7c     mflr    r0
   4:   f8 ff e1 fb     std     r31,-8(r1)
   8:   f0 ff c1 fb     std     r30,-16(r1)
   c:   e8 ff a1 fb     std     r29,-24(r1)
  10:   e0 ff 81 fb     std     r28,-32(r1)
  14:   10 00 01 f8     std     r0,16(r1)
  18:   00 00 e0 3b     li      r31,0
  1c:   78 1b 7e 7c     mr      r30,r3
  20:   78 23 9d 7c     mr      r29,r4
  24:   71 ff 21 f8     stdu    r1,-144(r1)
  28:   00 00 7e e8     ld      r3,0(r30)
  2c:   14 fa 63 7c     add     r3,r3,r31
  30:   00 00 7d f8     std     r3,0(r29)
  34:   08 00 bd 3b     addi    r29,r29,8
  38:   55 55 40 39     li      r10,21845
  3c:   c6 07 4a 79     rldicr  r10,r10,32,31
  40:   23 57 4a 65     oris    r10,r10,22307
  44:   36 12 4a 61     ori     r10,r10,4662
  48:   00 00 5d f9     std     r10,0(r29)
  4c:   08 00 bd 3b     addi    r29,r29,8
  50:   f0 ff 7d e8     ld      r3,-16(r29)
  54:   f8 ff 9d e8     ld      r4,-8(r29)
  58:   0a 00 a0 38     li      r5,10
  5c:   00 10 80 3d     lis     r12,4096
  60:   fc 0e 8c 61     ori     r12,r12,3836
  64:   a6 03 89 7d     mtctr   r12
  68:   21 1c 80 4e     bctrl   3
  6c:   78 fb e4 7f     mr      r4,r31
  70:   79 fb 6a 7c     or.     r10,r3,r31
  74:   08 00 82 40     bne     0x7c
  78:   01 00 80 38     li      r4,1
  7c:   f0 ff 9d f8     std     r4,-16(r29)
  80:   f8 ff bd 3b     addi    r29,r29,-8
  84:   f8 ff 3d e9     ld      r9,-8(r29)
  88:   00 f8 29 7c     cmpd    r9,r31
  8c:   28 00 82 41     beq     0xb4
  90:   78 fb e3 7f     mr      r3,r31
  94:   90 00 21 38     addi    r1,r1,144
  98:   10 00 01 e8     ld      r0,16(r1)
  9c:   e0 ff 81 eb     ld      r28,-32(r1)
  a0:   e8 ff a1 eb     ld      r29,-24(r1)
  a4:   f0 ff c1 eb     ld      r30,-16(r1)
  a8:   f8 ff e1 eb     ld      r31,-8(r1)
  ac:   a6 03 08 7c     mtlr    r0
  b0:   20 00 80 4e     blr
  b4:   01 00 60 38     li      r3,1
  b8:   90 00 21 38     addi    r1,r1,144
  bc:   10 00 01 e8     ld      r0,16(r1)
  c0:   e0 ff 81 eb     ld      r28,-32(r1)
  c4:   e8 ff a1 eb     ld      r29,-24(r1)
  c8:   f0 ff c1 eb     ld      r30,-16(r1)
  cc:   f8 ff e1 eb     ld      r31,-8(r1)
  d0:   a6 03 08 7c     mtlr    r0
  d4:   20 00 80 4e     blr
```

**Comment by zherczeg** [comment]
At first sight I would say the problem is here:
`sljit_emit_op2u(compiler, SLJIT_OR | SLJIT_SET_Z,
      SLJIT_RETURN_REG, 0, SLJIT_IMM, 0);`
Old code:
```
68:   79 fb 6a 7c     or.     r10,r3,r31
```
This is correct since r31 is zero (in our implementation), and set flags.
New code:
```
  68:   00 00 6a 60     ori     r10,r3,0
  6c:   00 00 4a 65     oris    r10,r10,0
```
This does not make any sense at all. Does not set flags, and emits two instructions. I will check it.

**Comment by zherczeg** [comment]
It looks like the code always was wrong, it was simply a luck that you used immediate 0, and that was translated to r31 earlier than in the current code. If you would use immediate 1, you would get a wrong code.

```
--- a/sljit_src/sljitNativePPC_common.c
+++ b/sljit_src/sljitNativePPC_common.c
@@ -1631,7 +1631,7 @@ SLJIT_API_FUNC_ATTRIBUTE sljit_s32 sljit_emit_op2(struct sljit_compiler *compile
                                return emit_op(compiler, GET_OPCODE(op), flags | ALT_FORM2, dst, dstw, src2, src2w, TMP_REG2, 0);
                        }
                }
-               if (GET_OPCODE(op) != SLJIT_AND) {
+               if (!HAS_FLAGS(op) && GET_OPCODE(op) != SLJIT_AND) {
                        /* Unlike or and xor, the and resets unwanted bits as well. */
                        if (TEST_UI_IMM(src2, src2w)) {
                                compiler->imm = (sljit_ins)src2w;
```

Does this change fixes it?

**Comment by zherczeg** [comment]
Anyway this code:
```
  sljit_emit_op1(compiler, SLJIT_MOV, SLJIT_R1, 0,
      SLJIT_IMM, 0);

  sljit_emit_op2u(compiler, SLJIT_OR | SLJIT_SET_Z,
      SLJIT_RETURN_REG, 0, SLJIT_IMM, 0);

  sljit_emit_cmov(compiler, SLJIT_ZERO, SLJIT_R1, SLJIT_IMM, 1);
```
Could probably be simplified to:
```
  // Compare (sub with no result) instruction is always preferred, systems has more instruction forms for it than 'or'
  sljit_emit_op2u(compiler, SLJIT_SUB | SLJIT_SET_Z, SLJIT_RETURN_REG, 0, SLJIT_IMM, 0);
  sljit_emit_op_flags(compiler, SLJIT_MOV, SLJIT_R1, 0, SLJIT_ZERO));
```

**Comment by cohomology** [comment]
Thanks! That was quick help!

The unit tests seem to run now (for me). I will see on monday the final central test run on linuxppc64 little endian and AIX big endian.

What I was missing most in sljit was a possibility to "easily" test a register for zero. On amd64 there are even specialized instructions for that (e.g. "test"). It is such a common task that `SLJIT_SUB | SLJIT_SET_Z` seems a bit overcomplicated and will also not generate the perfect instruction combination.

**Comment by zherczeg** [comment]
`test` can be generated by `op2u` with `SLJIT_AND | SLJIT_SET_Z`. I think this is efficient on all CPUs, since it is such a common operation. Or you can use `sljit_emit_cmp` if branching is also needed. Compare with 0 and branch is available on ARM64, MIPS and RISCV. I am not sure there is much perf difference between `test reg, reg`, or `cmp reg, 0`, but there might be.

**Comment by cohomology** [comment]
Ok, this night all unit tests went green on both platforms. 

Do you fix it in the upstream release and write a test?

If you want, you can then close the ticket. I will change my coding according to you recommendation though. I did first not really understand the semantics of `sljit_emit_op_flags` but I do now.

Regards
Kilian

**Comment by zherczeg** [comment]
Fixed in 3e86c50e9e4ae90f23cec8df6f25a7279e7ba6ea
